volume-providers/vpc/provider/create_volume_test.go

```diff
@@ -0,0 +1,224 @@
+/*******************************************************************************
+ * IBM Confidential
+ * OCO Source Materials
+ * IBM Cloud Container Service, 5737-D43
+ * (C) Copyright IBM Corp. 2018 All Rights Reserved.
+ * The source code for this program is not  published or otherwise divested of
+ * its trade secrets, irrespective of what has been deposited with
+ * the U.S. Copyright Office.
+ ******************************************************************************/
+
+package provider
+
+import (
+	"errors"
+	"github.com/IBM/ibmcloud-storage-volume-lib/lib/provider"
+	"github.com/IBM/ibmcloud-storage-volume-lib/lib/utils"
+	"github.com/IBM/ibmcloud-storage-volume-lib/lib/utils/reasoncode"
+	"github.com/IBM/ibmcloud-storage-volume-lib/volume-providers/vpc/vpcclient/models"
+	volumeServiceFakes "github.com/IBM/ibmcloud-storage-volume-lib/volume-providers/vpc/vpcclient/vpcvolume/fakes"
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/zap"
+	"testing"
+)
+
+func TestCreateVolume(t *testing.T) {
+	//var err error
+	logger, teardown := GetTestLogger(t)
+	defer teardown()
+
+	var (
+		volumeService *volumeServiceFakes.VolumeService
+	)
+
+	testCases := []struct {
+		name           string
+		baseVolume     *models.Volume
+		providerVolume provider.Volume
+		profileName    string
+
+		setup func(providerVolume *provider.Volume)
+
+		skipErrTest        bool
+		expectedErr        string
+		expectedReasonCode string
+
+		verify func(t *testing.T, volumeResponse *provider.Volume, err error)
+	}{
+		{
+			name: "Volume capacity is nil",
+			providerVolume: provider.Volume{
+				VolumeID: "16f293bf-test-4bff-816f-e199c0c65db5",
+				Name:     String("Test volume"),
+			},
+			baseVolume: &models.Volume{
+				ID:       "16f293bf-test-4bff-816f-e199c0c65db5",
+				Name:     "test volume name",
+				Status:   models.StatusType("OK"),
+				Capacity: int64(10),
+				Iops:     int64(1000),
+				Zone:     &models.Zone{Name: "test-zone"},
+			},
+			verify: func(t *testing.T, volumeResponse *provider.Volume, err error) {
+				assert.Nil(t, volumeResponse)
+				assert.NotNil(t, err)
+			},
+		}, {
+			name: "Volume name is nil",
```

Comment:
In case of volume name nil then provider.CreateVolume method will not call the vpcclient.CreateVolume hence baseVolume is not required for test case even providerVolume should be nil.

are we verifying error code which is returned by CreateVolume or other methods 

applicables for other places as well

Reply:
Yeah...we are doing that.

By using fakes, we are passing the base volume object to provider create volume function and that is checking the volume object. That is why we are getting coverage.